frontend/apps/crates/entry/jig/edit/src/edit/sidebar/menu/dom.rs

```diff
@@ -18,157 +18,199 @@ const STR_PASTE: &'static str = "Paste from another JIG";
 const STR_DUPLICATE_AS: &'static str = "Duplicate content as:";
 // const STR_EDIT_SETTINGS: &'static str = "Edit setting";
 
-pub fn render(state: Rc<State>, items: Vec<Dom>) -> Dom {
-    html!("menu-kebab", {
-        .property("slot", "menu")
-        .child(html!("jig-edit-sidebar-module-menu", {
-            .children(items)
-        }))
-        .event_with_options(&EventOptions::bubbles(), |e: events::Click| {
-            e.stop_propagation();
+pub struct MenuDom;
+
+impl MenuDom {
+    pub fn render(module_state: Rc<ModuleState>) -> Dom {
```

Comment:
@johnnynotsolucky why the move to a struct with a method?

Reply:
@MendyBerger only to get it more or less similar to how other components are built, for example [EditPage](https://github.com/johnnynotsolucky/ji-cloud/blob/sandbox/frontend/apps/crates/entry/jig/edit/src/edit/dom.rs#L17-L20) and [ModuleDom](https://github.com/johnnynotsolucky/ji-cloud/blob/sandbox/frontend/apps/crates/entry/jig/edit/src/edit/sidebar/module/dom.rs#L15-L18). 

Which is preferred for these?

If I should revert, it's minimal work to revert it.

Reply:
Also, these refactors should've been in a separate PR to keep PRs simpler. I'll do that next time.

Reply:
The way components are structured has changed over time, having a Dom struct that has a method was actually before the bare function, but we have now moved on from the bare function as well.

Maybe I can give you a basic rundown of this in the daily meeting.

Reply:
I reverted most of the refactor of MenuDom, I'll make a separate PR later with more up-to-date changes.